src/elm/Target/Models.elm

```diff
@@ -1,33 +1,30 @@
-module Target.Models exposing (..)
+module Target.Models
+    exposing
+        ( Target
+        , initial
+        , currentTarget
+        )
 
-import Maybe exposing (withDefault)
+import Data.DataSource exposing (DataSource, DataSourceId(..))
 
 
 type alias Target =
-    { all : DataSources
-    , current : Int
+    { current : Maybe DataSourceId
     , filter : String
     }
 
 
-type alias DataSources =
-    List DataSource
+initial : Target
+initial =
+    { current = Nothing, filter = "" }
 
 
-type alias DataSource =
-    { id : Int
-    , title : Maybe String
-    , desc : Maybe String
-    }
-
-
-initTarget : List Int -> Target
-initTarget dss =
-    let
-        current =
-            withDefault 1 <| List.head dss
+currentTarget : List DataSource -> Target -> Maybe DataSource
+currentTarget dataSources { current } =
+    case current of
+        Nothing ->
+            Nothing
```

Comment:
It looks like you're using the `case` her to for flattening purposes? Thoughts on using `Maybe.andThen` instead so you don't need to say `Nothing -> Nothing` ?